src/zenml/orchestrators/step_run_utils.py

```diff
@@ -236,6 +247,30 @@ def _try_to_get_docstring_and_source_code_from_template(
 
         return None, None
 
+    def _define_input_type(
+        self, input_name: str, step: Step
+    ) -> StepRunInputArtifactType:
+        """Define the input type of an artifact.
+
+        Args:
+            input_name: The name of the input artifact.
+            step: The step instance.
+
+        Returns:
+            The input type of the artifact.
+        """
+        if input_name in step.spec.inputs:
+            return StepRunInputArtifactType.STEP_OUTPUT
+        elif input_name in step.config.external_input_artifacts:
+            return StepRunInputArtifactType.EXTERNAL
+        elif (
+            input_name in step.config.model_artifacts_or_metadata
+            or input_name in step.config.client_lazy_loaders
+        ):
+            return StepRunInputArtifactType.LAZY_LOADED
+        else:
+            return StepRunInputArtifactType.MANUAL
```

Comment:
This should not be possible, an artifact can't be manually loaded inside the step if the step didn't actually run yet. Probably best to fail here instead